File: push-client-java/src/main/java/wang/ismy/push/client/factory/SocketFactory.java
package wang.ismy.push.client.factory;

import java.io.IOException;
import java.net.Socket;

/**
 * @author MY
 * @date 2020/9/15 17:03
 */
public class SocketFactory {

    public Socket newSocket(String host, int port) throws IOException {
        return new Socket(host,port);
    }

}

File: push-clients/push-client-javascript/socket_factory.js

const net = require('net')

async function newSocket(host, port){
  return new Promise((resolve, rejects) => {
    const client = net.connect(
      {host, port},
      (e) => {
        resolve(client)
      }
    )
  })
}

module.exports = { newSocket }